### DISCUSSION TITLE
Wavefield Discrepancy between Finite Difference Modeling and Analytical Results

### DISCUSSION THREAD

**Discussion by tjk9501** [slimgroup/JUDI.jl#228] [Q&A] [answered]
Hello,

I am using full waveform inversion to run some forward modeling simulations. And I am checking the data matching between finite difference simulations and analytical results. Here, I use the zeroth order and second type Hankel function, conj(\frac{i}{4}H_{0}^{2}(k\left | \mathbf{r-r_{0}} \right |))
![CodeCogsEqn](https://github.com/slimgroup/JUDI.jl/assets/54341310/81de1e5d-dcf7-478a-b86a-32ec61350135)
and I also checked -\frac{i}{4}H_{0}^{1}(k\left | \mathbf{r-r_{0}} \right |) 
![CodeCogsEqn (1)](https://github.com/slimgroup/JUDI.jl/assets/54341310/f2e5f465-f191-47ed-b726-7164d58e9c62)
as analytical results, where \mathbf{r_{0}}
![CodeCogsEqn (2)](https://github.com/slimgroup/JUDI.jl/assets/54341310/be6b29d5-cb81-4fba-b04b-280cfec2be39)
denote the position of source point. The first used is a circular array setup, with model size being 1000*1000 mm and grid spacing of 2 mm,  as shown in the following figure
![1-1](https://github.com/slimgroup/JUDI.jl/assets/54341310/77ee2614-10a2-4cf2-b5ff-4fe0ea593e07)
The resulting wavefield, converted into frequency domain and obtained by finite difference simulations (FDM), is shown in the following
![1-2](https://github.com/slimgroup/JUDI.jl/assets/54341310/7b6609a0-672b-4d9a-9fea-9ee5520ad94b)
where ith index in the x axis denote the data obsered by ith transducer. We can see good matching between FDM results and hankel functions (Theory).

Then I increased the model size into 2000 * 3000 mm, using a different array setup
![1-3](https://github.com/slimgroup/JUDI.jl/assets/54341310/e57513d2-7a6c-4a0b-9faa-61acc6ea5663)

The resulting wavefield is
![1-4](https://github.com/slimgroup/JUDI.jl/assets/54341310/83903447-206f-4a86-a124-1fa9f80aa9ed)
It shows great mismatch between theory and FDM methods.

To check whether this effect is caused by grid size, I reduced the model size into 2000 * 1004 mm, and use a denser grid spacing of 1 mm. The resulting wavefield is
![1mm_Spacing](https://github.com/slimgroup/JUDI.jl/assets/54341310/f4d7d6d5-54e4-4f56-858a-343e1bc99a00)

It still shows great mismatch. Why is such phenomenon occured? What is the possible cause of this effect? Is it possible to correct the analytical Hankel function to compensate for such error in simulations?

The above results is obtained with the open source software fastFWI. However, using JUDI I also find similar problems, here I present the comparison between JUDI results with hankel function
real part:
![JUDI_1_real](https://github.com/slimgroup/JUDI.jl/assets/54341310/f1297c66-4b3d-4f70-a784-f61ae4978219)
imaginary part:
![JUDI_1_imag](https://github.com/slimgroup/JUDI.jl/assets/54341310/cbf9cc29-eb85-4f87-821b-a30f072716eb)

Despite the amplitude difference, phase mismatch seems more important.

Thanks!

**Comment by mloubout** [comment, marked as answer]
Hi

Thanks for reaching out. I am not quite sure what the setup/script/... is here but I do know that JUDI matches analytical solution, the script checking it is in devito:

https://nbviewer.org/github/devitocodes/devito/blob/master/examples/seismic/acoustic/accuracy.ipynb

I may try to make a similar notebook in JUDI if I find time. 

Some general comments

- I would only use a single source and single receiver as adding more only add to the complexity of the test while a single source gives you all you need for an accuracy test
- I would avoid comparing in the frequency domain to avoid any edge effect/...
- I am not sure what source wavelet you are using but need to make sure the peak frequency is low enough to avoid any time dispersion

I hope this helps, I'm happy to discuss it more if needed.

**Reply by tjk9501** [reply]
Thanks, I will check my simulations to see where goes wrong.
